server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -0,0 +1,257 @@
+        for i in sorted_s3_content[i:len_s3_content]:
+            report.write(
+                "{}: present in s3 but not in local backup\n".format(i.name))
+
+
+def report_primary_backup_s3(reference_dir, report):
+    # Function to compare archive, backup and s3
+    aname = os.path.join(reference_dir, "list.archive")
+    try:
+        with open(aname) as f:
+            primary_list = set(f.readlines())
+    except Exception as e:
+        _logger.exception("error reading {}", aname)
+        return
+
+    bname = os.path.join(reference_dir, "list.backup")
+    try:
+        with open(bname) as f:
+            backup_list = set(f.readlines())
+    except Exception as e:
+       _logger.exception("error reading {}", bname)
+       return
+
+    sname = os.path.join(reference_dir, "list.s3")
+    try:
+        with open(sname) as f:
+            s3_list = set(f.readlines())
+    except Exception as e:
+        _logger.exception("error reading {}", sname)
+        return
+
+    only_p = primary_list.difference(backup_list, s3_list)
+    for i in only_p:
+        report.write("{}: only in archive\n".format(i.strip('\n')))
+
+    only_b = backup_list.difference(primary_list, s3_list)
+    for j in only_b:
+        report.write("{}: only in backup\n".format(j.strip('\n')))
+
+    only_s3 = s3_list.difference(primary_list, backup_list)
+    for k in only_s3:
+        report.write("{}: only in s3\n".format(k.strip('\n')))
+
+
+def main():
+    cfg_name = os.environ.get("CONFIG")
+    if not cfg_name:
+        print("{}: ERROR: No config file specified; set CONFIG env variable or"
+                " use --config <file> on the command line".format(_NAME_),
+                file=sys.stderr)
+        return 2
+
+    try:
+        config = PbenchConfig(cfg_name)
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 1
+
+    global _logger
+    _logger = get_pbench_logger(_NAME_, config)
+
+    archive = config.ARCHIVE
+    if not os.path.isdir(archive):
+        _logger.error(
+            'The ARCHIVE directory does not resolve to a directory, {}'.format(archive))
+        return 1
+
+    # add a BACKUP field to the config object
+    config.BACKUP = backup = config.conf.get(
+        "pbench-server", "pbench-backup-dir")
+    if len(backup) == 0:
+        _logger.error(
+            'Unspecified backup directory, no pbench-backup-dir config in pbench-server section')
+        return 1
+
+    if not os.path.isdir(backup):
+        _logger.error('Specified backup directory, {}, does not resolve {} to a directory'.format(
+            backup, os.path.realpath(backup)))
+        return 1
+
+    _logger.info('start-{}'.format(config.timestamp()))
+
+    prog = os.path.basename(sys.argv[0])
+    with tempfile.TemporaryDirectory() as reference_dir:
+        # Check the data integrity in Archive.
+        checkmd5(config.ARCHIVE, reference_dir, "archive")
```

Comment:
This should return status, so check it here. I think it still makes sense to continue on a failure, but you might need to remember that this failed, so you can do only the things that make sense in that case.